Sitemap/Url/GoogleVideoUrlDecorator.php

```diff
@@ -198,58 +204,88 @@ class GoogleVideoUrlDecorator extends UrlDecorator
      *
      * @throws Exception\GoogleVideoUrlException
      */
-    public function __construct(Url $urlDecorated, $thumnail_loc, $title, $description, array $parameters = [])
+    public function __construct(Url $urlDecorated, $thumnail_loc = null, $title = null, $description = null, array $parameters = null)
     {
-        foreach ($parameters as $key => $param) {
-            $method = Utils::getSetMethod($this, $key);
-            $this->$method($param);
-        }
+        parent::__construct($urlDecorated);
 
-        $this->setThumbnailLoc($thumnail_loc);
-        $this->setTitle($title);
-        $this->setDescription($description);
+        if ($thumnail_loc !== null || $title !== null || $description !== null || $parameters !== null) {
+            @trigger_error('Using other arguments than $urlDecorated in constructor is deprecated. Create a GoogleVideo object instead.', E_USER_DEPRECATED);
 
-        if (!$this->content_loc && !$this->player_loc) {
-            throw new Exception\GoogleVideoUrlException('The parameter content_loc or player_loc is required');
+            $this->video = new GoogleVideo($thumnail_loc, $title, $description, $parameters ?: []);
         }
+    }
 
-        if (count($this->platforms) && !$this->platform_relationship) {
-            throw new Exception\GoogleVideoUrlException(
-                'The parameter platform_relationship is required when platform is set'
-            );
+    /**
+     * Checker and deprecation triggerer for backward compatibility
+     *
+     * @param string $methodName
+     */
+    private function bc(string $methodName): void
+    {
+        switch (substr($methodName, 0, 3)) {
+            case 'set':
+                $text = 'Using %s is deprecated. Create a GoogleVideo object instead.';
+                break;
+            case 'get':
+                $text = 'Using %s is deprecated. Retrieve it from GoogleVideo object instead.';
+                break;
+            case 'add':
+                $text = 'Using %s is deprecated. Add it to GoogleVideo object instead.';
+                break;
+            default:
+                $text = 'Using %s is deprecated.';
         }
 
-        parent::__construct($urlDecorated);
+        @trigger_error(
+            sprintf($text, $methodName),
+            E_USER_DEPRECATED
+        );
```

Comment:
Can you add `__CLASS__` to all messages, so the FQCN is included in error logs ?

- `Using %s::%s is deprecated. Create a GoogleVideo object instead.`
- `Using %s::%s is deprecated. Retrieve it from GoogleVideo object instead.`
- `Using %s::%s is deprecated. Add it to GoogleVideo object instead.`
- `Using %s::%s is deprecated.`

`@trigger_error(sprintf($text, __CLASS__, $methodName), E_USER_DEPRECATED);`